updater/aws.go

```diff
@@ -60,6 +62,37 @@ type EC2API interface {
 	WaitUntilInstanceStatusOk(input *ec2.DescribeInstanceStatusInput) error
 }
 
+func (u *updater) alreadyRunning(family string) (bool, error) {
+	log.Print("Checking for running updater tasks")
+	list, err := u.ecs.ListTasks(&ecs.ListTasksInput{
+		Cluster: &u.cluster,
+		Family:  aws.String(family),
+	})
+	if err != nil {
+		return false, fmt.Errorf("failed to list running updater tasks: %w", err)
+	}
+	if len(list.TaskArns) > 1 {
+		return true, nil
+	}
+	log.Println("This is the only running updater.")
+	return false, nil
+}
+
+func taskDefFamily() (string, error) {
+	taskDefARN := os.Getenv("TaskDefinitionARN")
+	taskDefARNType, err := arn.Parse(taskDefARN)
```

Comment:
nit: I'd rather not have "type" in the variable name since the variable isn't holding a type
```suggestion
	taskDefInput := os.Getenv("TASK_DEFINITION_ARN")
	taskDefARN, err := arn.Parse(taskDefInput)
```